pontoon/base/models.py

```diff
@@ -2872,28 +2874,139 @@ def serialize(self):
+        sql_substring = 'SUBSTRING(source, 0, %s)' % substring_filter_size
+
+        possible_matches = (
+            self.extra(
+                where=[
+                    '(CHAR_LENGTH(source) BETWEEN %s AND %s)',
+                    self.get_levenshtein_sql(sql_substring) + ' > %s',
+                ],
+                params=(min_dist, max_dist, len(subtext), subtext, len(subtext), min_quality),
+            )
+            .values_list('pk', 'source')
+            .filter(locale=locale)
+        )
+
+        matches_pks = []
+
+        # In order to keep compatibility with `postgresql_levenshtein_ratio`,
+        # entries are annotate with the quality column.
+        quality_sql_map = []
+
+        for pk, source in possible_matches:
+            quality = Levenshtein.ratio(text, source)
+
+            if quality > min_quality:
+                matches_pks.append(pk)
+                quality_sql_map.append(
+                    When(
+                        pk=pk,
+                        then=Value(quality * 100)
+                    )
+                )
+        entries = (
+            self.filter(
+                pk__in=matches_pks,
+            ).annotate(
+                quality=Case(
+                    *quality_sql_map,
+                    **dict(
+                        default=Value(0),
+                        output_field=IntegerField(),
+                    )
+                )
+            )
+        )
+        return entries
+
+    def minimum_levenshtein_ratio(self, locale, text, min_quality=0.7):
         """
         Returns entries that match minimal levenshtein_ratio
         """
+        # Only check entities with similar length
         length = len(text)
         min_dist = math.ceil(max(length * min_quality, 2))
         max_dist = math.floor(min(length / min_quality, 1000))
-        levenshtein_ratio_equation = """(
-            (char_length(source) + char_length(%s) - levenshtein(source, %s, 1, 2, 2))::float /
-            (char_length(source) + char_length(%s))
-        )"""
 
-        # Only check entities with similar length
-        entries = self.extra(
-            where=[
-                '(CHAR_LENGTH(source) BETWEEN %s AND %s)',
-                levenshtein_ratio_equation + ' > %s'
-            ],
-            params=(min_dist, max_dist, text, text, text, min_quality),
-            select={'quality': levenshtein_ratio_equation + '* 100'},
-            select_params=(text, text, text)
-        )
+        if min_dist > 255 or max_dist > 255:
+            entries = self.python_levenshtein_ratio(
+                locale,
+                text,
+                min_quality,
+                min_dist,
+                max_dist,
+            )
+        else:
+            entries = self.postgres_levenshtein_ratio(
+                locale,
+                text,
+                min_quality,
+                min_dist,
+                max_dist,
+            )
+
         return entries
```

Comment:
This could be simplified to something like this, for less repetition:

```python
get_matches = self.postgres_levenshtein_ratio
if min_dist > 255 or max_dist > 255:
    get_matches = self.python_levenshtein_ratio

return get_matches(
     locale,
     text,
     min_quality,
     min_dist,
     max_dist,
)
```